src/Servers/Kestrel/Core/src/Internal/Infrastructure/TransportManager.cs

```diff
@@ -66,6 +66,17 @@ public async Task<EndPoint> BindAsync(EndPoint endPoint, MultiplexedConnectionDe
                     ApplicationProtocols = new List<SslApplicationProtocol>() { new SslApplicationProtocol("h3"), new SslApplicationProtocol("h3-29") }
                 };
 
+                if (listenOptions.HttpsOptions.ServerCertificateSelector != null)
+                {
+                    // We can't set both
+                    sslServerAuthenticationOptions.ServerCertificate = null;
```

Comment:
Is automatically setting ServerCertificate to null what HTTP/1.1 and HTTP/2 does?

Reply:
Yes:
https://github.com/dotnet/aspnetcore/blob/fa9dd9c5fe476706e33819ded8b96bf72a5fef41/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs#L89-L92